src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -28,10 +28,13 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
         {
             if (workspaceServices.Workspace is VisualStudioWorkspace vsWorkspace)
             {
-                var experimentationService = vsWorkspace.Services.GetService<IExperimentationService>();
-                if (!experimentationService.IsExperimentEnabled(WellKnownExperimentNames.PartialLoadMode))
+                var experimentationServiceFactory = vsWorkspace.Services.GetService<IExperimentationServiceFactory>();
+                var experimentationServiceTask = experimentationServiceFactory.GetExperimentationServiceAsync(CancellationToken.None);
+                if (experimentationServiceTask.IsCompleted && !experimentationServiceTask.Result.IsExperimentEnabled(WellKnownExperimentNames.PartialLoadMode))
                 {
-                    // don't enable partial load mode for ones that are not in experiment yet
+                    // Don't enable partial load mode for ones that are not in experiment yet. If the experimentation
+                    // service isn't available by the time this is called, the VS Service instance is used and it will
+                    // check the value of the experiment at the time individual features are needed.
```

Comment:
I don't understand this comment.

Reply:
I'm having trouble with the wording. This is an opportunistic fast-path. We _try_ to check the experimentation service, and if that succeeds and the experiment is not enabled, we return a lighter instance of `IWorkspaceStatusService`. If we can't check the experiment here, we use `VisualStudioWorkspaceStatusServiceFactory.Service`, and each call to that service will check the experiment.